pkg/statistics/cmsketch.go

```diff
@@ -520,13 +522,23 @@ func (c *CMSketch) CalcDefaultValForAnalyze(ndv uint64) {
 // TopN stores most-common values, which is used to estimate point queries.
 type TopN struct {
 	TopN []TopNMeta
+
+	totalCount uint64
+	minCount   uint64
+	// minCount and totalCount are initialized only once.
+	once sync.Once
 }
 
-// Scale scales the TopN by the given factor.
-func (c *TopN) Scale(scaleFactor float64) {
+// Scale scales the TopN by the given factor and return the total row count.
+// It is only used when to build topN from sampled data during analyze.
+func (c *TopN) Scale(scaleFactor float64) uint64 {
```

Comment:
Do we need to call "calculateMinCountAndCount" again after this Scale?

Reply:
No, The `calculateMinCountAndCount` in ' scale ' is actually not necessary. The subsequent logic doesn't need `TotalCount` anymore. It just writes directly to TiKV.In fact, `scale` is only used in that one place, so we could actually remove the `scale` function and integrate it directly into the ' analyze ' part.